### ISSUE TITLE
when Proton is selected rare crashes on startup

### ISSUE THREAD

**Issue by MrSwedish** [Dummerle/Rare#83]
arch linux AUR git version

**Comment by Dummerle** [comment]
Please add more information like logs

**Comment by MrSwedish** [comment]
oh right, forgot here you go:
```
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/rare/components/tabs/cloud_saves/__init__.py", line 46, in <lambda>
    self.start_thread.signal.connect(lambda x: self.setup_ui(x, app_name, auto))
  File "/usr/lib/python3.9/site-packages/rare/components/tabs/cloud_saves/__init__.py", line 84, in setup_ui
    sync_widget = SyncWidget(igame, latest_save[igame.app_name], self.core)
  File "/usr/lib/python3.9/site-packages/rare/components/tabs/cloud_saves/sync_widget.py", line 85, in __init__
    if not os.path.exists(igame.save_path):
  File "/usr/lib/python3.9/genericpath.py", line 19, in exists
    os.stat(path)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType
Aborted (core dumped)
```

**Comment by Dummerle** [comment]
I fixed it, but in shop branch, because i'm working on shop

**Comment by Dummerle** [comment]
You can try it again in default branch

**Comment by MrSwedish** [comment]
alright, let me see

**Comment by MrSwedish** [comment]
still crashes
```
Traceback (most recent call last):
  File "/home/kaktus/testRARE/Rare/rare/components/tabs/cloud_saves/__init__.py", line 46, in <lambda>
    self.start_thread.signal.connect(lambda x: self.setup_ui(x, app_name, auto))
  File "/home/kaktus/testRARE/Rare/rare/components/tabs/cloud_saves/__init__.py", line 84, in setup_ui
    sync_widget = SyncWidget(igame, latest_save[igame.app_name], self.core)
  File "/home/kaktus/testRARE/Rare/rare/components/tabs/cloud_saves/sync_widget.py", line 85, in __init__
    if not os.path.exists(igame.save_path):
  File "/usr/lib/python3.9/genericpath.py", line 19, in exists
    os.stat(path)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType
./start.sh: line 2: 31963 Aborted                 (core dumped) python3 rare
```
cloned from master/main

**Comment by Dummerle** [comment]
I know what's the mistake. I will fix it, when I am at my pc

**Comment by Dummerle** [comment]
You can try it again

**Comment by MrSwedish** [comment]
alrighty lets try

**Comment by MrSwedish** [comment]
doesn't crash but nothing runs with proton, even looking into protondb and trying out fixes didn't work
even platinum games didn't work

**Comment by Dummerle** [comment]
Which Proton version do you use?

**Comment by MrSwedish** [comment]
tried all of them, even ge-custom

**Comment by MrSwedish** [comment]
and sometimes it crashes when starting a game with proton and has no log in the console

**Comment by MrSwedish** [comment]
when starting just legendary this happens:
```
Traceback (most recent call last):
  File "/home/kaktus/.steam/steam/steamapps/common/Proton 6.3/proton", line 987, in <module>
    g_session.init_session(sys.argv[1] != "runinprefix")
  File "/home/kaktus/.steam/steam/steamapps/common/Proton 6.3/proton", line 816, in init_session
    g_compatdata.setup_prefix()
  File "/home/kaktus/.steam/steam/steamapps/common/Proton 6.3/proton", line 419, in setup_prefix
    with self.prefix_lock:
  File "/home/kaktus/.local/share/Steam/steamapps/common/Proton 6.3/filelock.py", line 323, in __enter__
    self.acquire()
  File "/home/kaktus/.local/share/Steam/steamapps/common/Proton 6.3/filelock.py", line 271, in acquire
    self._acquire()
  File "/home/kaktus/.local/share/Steam/steamapps/common/Proton 6.3/filelock.py", line 384, in _acquire
    fd = os.open(self._lock_file, open_mode)
FileNotFoundError: [Errno 2] No such file or directory: '/home/kaktus/.proton/pfx.lock'
```
I get a log, maybe this is legendary's issue maybe the configs get set wrong from rare, just look at it

**Comment by MrSwedish** [comment]
oh and when saving the wrapper options of a game, it crashed (don't know if it always does that)
```
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/rare/components/tabs/games/game_info/game_settings.py", line 63, in <lambda>
    lambda: self.save_line_edit("wrapper", self.wrapper.text())
  File "/usr/lib/python3.9/site-packages/rare/components/tabs/games/game_info/game_settings.py", line 92, in save_line_edit
    if not self.core.lgd.config[self.game.app_name]:
  File "/usr/lib/python3.9/configparser.py", line 960, in __getitem__
    raise KeyError(key)
```

**Comment by Dummerle** [comment]
> when starting just legendary this happens:
> 
> ```
> Traceback (most recent call last):
>   File "/home/kaktus/.steam/steam/steamapps/common/Proton 6.3/proton", line 987, in <module>
>     g_session.init_session(sys.argv[1] != "runinprefix")
>   File "/home/kaktus/.steam/steam/steamapps/common/Proton 6.3/proton", line 816, in init_session
>     g_compatdata.setup_prefix()
>   File "/home/kaktus/.steam/steam/steamapps/common/Proton 6.3/proton", line 419, in setup_prefix
>     with self.prefix_lock:
>   File "/home/kaktus/.local/share/Steam/steamapps/common/Proton 6.3/filelock.py", line 323, in __enter__
>     self.acquire()
>   File "/home/kaktus/.local/share/Steam/steamapps/common/Proton 6.3/filelock.py", line 271, in acquire
>     self._acquire()
>   File "/home/kaktus/.local/share/Steam/steamapps/common/Proton 6.3/filelock.py", line 384, in _acquire
>     fd = os.open(self._lock_file, open_mode)
> FileNotFoundError: [Errno 2] No such file or directory: '/home/kaktus/.proton/pfx.lock'
> ```
> 
> I get a log, maybe this is legendary's issue maybe the configs get set wrong from rare, just look at it

You need to create .proton folder. Then should it work

The other error should be fixed

**Comment by MrSwedish** [comment]
alrighty, let me try again

**Comment by MrSwedish** [comment]
you could create a `.proton` folder with `mkdir -p ~/.proton`

**Comment by MrSwedish** [comment]
worked, i hope it used proton
thanks

**Comment by Dummerle** [comment]
Nice
